pkg/app/bootstrap.go

```diff
@@ -289,14 +371,33 @@ func buildkit(clicontext *cli.Context) error {
 	}
 
 	logrus.Debug("bootstrap the buildkitd container")
-	var bkClient buildkitd.Client
-	config := buildkitutil.BuildkitConfig{
-		Registry: clicontext.String("registry"),
-		Mirror:   clicontext.String("dockerhub-mirror"),
-		UseHTTP:  clicontext.Bool("use-http"),
-		SetCA:    clicontext.IsSet("registry-ca-keypair"),
+	// Populate the BuildkitConfig struct
+	config := buildkitutil.BuildkitConfig{}
+
+	configFile := clicontext.String("registry-config")
+	if len(configFile) != 0 {
+		configJson, err := os.ReadFile(configFile)
+		if err != nil {
+			return errors.Wrap(err, "Failed to read registry config file")
+		}
+		if err := json.Unmarshal(configJson, &config); err != nil {
+			return errors.Wrap(err, "Failed to parse registry config file")
+		}
+	} else if len(clicontext.String("registry-ca-keypair")) != 0 {
+		config.Registries = append(config.Registries, buildkitutil.Registry{
+			Name:    clicontext.String("registry"),
+			Ca:      "/etc/registry",
+			Cert:    "/etc/registry",
+			Key:     "/etc/registry",
+			UseHttp: clicontext.Bool("use-http"),
+			Mirror:  clicontext.String("dockerhub-mirror"),
+		})
+		// The values of Ca, Cert, and Key don't actually matter since we already copied their contents to /etc/registry
+		// So instead of parsing registry-ca-keypair again, we'll just put an arbitrary value
+		// This is to ensure that buildkitConfigTemplate parses properly
```

Comment:
```suggestion
		// The values of Ca, Cert, and Key don't actually matter since we already copied their contents to the envd config directory and mounted to `/etc/registry`.
		// So instead of parsing registry-ca-keypair again, we'll just put the default value.
		// This is to ensure that buildkitConfigTemplate parses properly.
		config.Registries = append(config.Registries, buildkitutil.Registry{
			Name:    clicontext.String("registry"),
			Ca:      "/etc/registry",
			Cert:    "/etc/registry",
			Key:     "/etc/registry",
			UseHttp: clicontext.Bool("use-http"),
			Mirror:  clicontext.String("dockerhub-mirror"),
		})
```